src/Orleans.Runtime/Hosting/Generic/SiloHostBuilderExtensions.cs

```diff
@@ -108,5 +110,46 @@ public static ISiloHostBuilder ConfigureLogging(this ISiloHostBuilder builder, A
         {
             return builder.ConfigureServices(collection => collection.AddLogging(configureLogging));
         }
+
+        /// <summary>
+        /// Returns the <see cref="ApplicationPartManager"/> for this instance.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <returns>The <see cref="ApplicationPartManager"/> for this instance.</returns>
+        public static ApplicationPartManager GetApplicationPartManager(this ISiloHostBuilder builder) => ApplicationPartManagerExtensions.GetApplicationPartManager(builder.Properties);
+
+        /// <summary>
+        /// Adds an <see cref="IApplicationPart"/> to the <see cref="ApplicationPartManager"/>.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <param name="assembly">The assembly.</param>
+        /// <returns>The builder.</returns>
+        public static ISiloHostBuilder AddApplicationPart(this ISiloHostBuilder builder, Assembly assembly)
```

Comment:
I'm on the fence with this one. Do you envision this to be the most common case? Otherwise to make the jump to add references too for example, then it's a lot different, and maybe the best is to get them to use the fluent interface in the first place.

Reply:
I'm on the fence, too - should we just axe it? Shortcuts like this can be added later if we want, so maybe it's better to be conservative.

Reply:
My vote is yes: let's axe it for now

Reply:
Should I wait for this change or you think we should keep it then?